#### Se carga el data set 'iris-data.csv' y se observa su estructura.

In [1]:
import numpy as np
import pandas as pd

In [2]:
!curl https://raw.githubusercontent.com/plotly/datasets/master/iris-data.csv -o iris-data.csv

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  4757  100  4757    0     0  12831      0 --:--:-- --:--:-- --:--:-- 12856


In [3]:
iris_df = pd.read_csv('iris-data.csv')

In [4]:
print(iris_df.shape)

(150, 5)


In [5]:
print(iris_df[:5])
# Idealmente se realiza con iris_df.head()

   sepal length  sepal width  petal length  petal width        class
0           5.1          3.5           1.4          0.2  Iris-setosa
1           4.9          3.0           1.4          0.2  Iris-setosa
2           4.7          3.2           1.3          0.2  Iris-setosa
3           4.6          3.1           1.5          0.2  Iris-setosa
4           5.0          3.6           1.4          0.2  Iris-setosa


#### Se añaden lineas de código adicionales para realizar una descripción un poco más profunda del data set.

In [6]:
iris_df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal length  150 non-null    float64
 1   sepal width   150 non-null    float64
 2   petal length  150 non-null    float64
 3   petal width   150 non-null    float64
 4   class         150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [7]:
iris_df.isnull().sum()

sepal length    0
sepal width     0
petal length    0
petal width     0
class           0
dtype: int64

In [8]:
print(iris_df['class'].unique())

['Iris-setosa' 'Iris-versicolor' 'Iris-virginica']


#### Observamos que el data set se compone de cuatro variables numéricas flotantes: "sepal length" (longitud del sépalo), "sepal width" (ancho del sépalo), "petal length" (longitud del pétalo), "petal width" (ancho del pétalo); y una variable categórica de 3 categorías ('Iris setosa', 'Iris versicolor' e 'Iris virginica', que corresponden a distintos tipos de flores). 

#### Dentro del data set hay 150 filas o casos observados y no se observan valores nulos. 

#### A continuación, observaremos los estadísticos para cada variable. 

In [9]:
print(iris_df.describe())

       sepal length  sepal width  petal length  petal width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.054000      3.758667     1.198667
std        0.828066     0.433594      1.764420     0.763161
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000


#### En general, podríamos decir que el sépalo tiene una mayor longitud y ancho que los pétalos: El sépalo posee una longitud promedio de 5.84 cm y un ancho promedio de 3.1 cm, en contraste al pétalo que posee una longitud promedio de 3.76 cm y un ancho promedio de 1.2 cm. 
#### Por otro lado, el pétalo posee una mayor disperción en contraste a los sépalos, por lo que el tamaño de éstos varía mucho más por caso observado. 

#### Demos un paso más y observemos cómo se comportan estos valores según tipo de flor: 

In [11]:
estadisticas_iris = iris_df.groupby('class').agg({
    'sepal length': ['count', 'mean', 'std'],
    'sepal width': ['count', 'mean', 'std'],
    'petal length': ['count', 'mean', 'std'],
    'petal width': ['count', 'mean', 'std']
})

print(estadisticas_iris)

                sepal length                  sepal width                   \
                       count   mean       std       count   mean       std   
class                                                                        
Iris-setosa               50  5.006  0.352490          50  3.418  0.381024   
Iris-versicolor           50  5.936  0.516171          50  2.770  0.313798   
Iris-virginica            50  6.588  0.635880          50  2.974  0.322497   

                petal length                  petal width                   
                       count   mean       std       count   mean       std  
class                                                                       
Iris-setosa               50  1.464  0.173511          50  0.244  0.107210  
Iris-versicolor           50  4.260  0.469911          50  1.326  0.197753  
Iris-virginica            50  5.552  0.551895          50  2.026  0.274650  


#### De esta manera observamos en primer que existen 50 filas u observaciones para cada clase de flor. 
#### En segundo lugar, que el tipo de flor 'Iris virginica' posee la mayor longitud promedio de sépalo y pétalo, así como también el mayor ancho promedio de pétalo, por lo que sería la especie más grande de las 3. Estas características también aplican para su disperción, por lo que este tipo de flor tiene mayores variaciones de tamaño en cada caso de observación. 
#### También observamos que la especie 'Iris setosa' es la que tiene el pétalo más pequeño en promedio, mientras que su sépalo es el que tiene la menor longitud promedio (aunque el mayor ancho promedio) por lo que podría tratarse de la especie más pequeña de las 3.